lib/elastomer/client/tasks.rb

```diff
@@ -0,0 +1,169 @@
+        response.body
+      end
+
+      # Wait for the specified amount of time (10 seconds by default) for some task(s) to complete.
+      # Filters for task(s) to wait upon using same filter params as Tasks#get(params)
+      #
+      # params          - Hash of request params to include (mostly task filters in this context)
+      #
+      # Examples
+      #
+      # tasks.wait_for("30s", {:actions => "*reindex", :nodes => "DmteLdw1QmSgW3GZmjmoKA,DmteLdw1QmSgW3GZmjmoKB"})
+      #
+      # Returns the response body as a Hash when timeout expires or target tasks complete
+      def wait_for(timeout = "10s", params = {})
+        params_with_wait = params.merge({ :wait_for_completion => true, :timeout => timeout })
+        self.get(params_with_wait)
+      end
+
+      # Wait for the specified amount of time (10 seconds by default) for some task(s) to complete.
+      # Filters for task(s) to wait upon using same IDs and filter params as Tasks#get_by_id(params)
+      #
+      # node_id                 - the ID of the node on which the target task is hosted
+      # task_id                 - the ID of the task to wait on
+      # params                  - Hash of request params to include (mostly task filters in this context)
+      #
+      # Examples
+      #
+      # tasks.wait_by_id "DmteLdw1QmSgW3GZmjmoKA", 123, "15s"
+      # tasks.wait_by_id "DmteLdw1QmSgW3GZmjmoKA", 456, "30s", :actions => "*search"
+      #
+      # Returns the response body as a Hash when timeout expires or target tasks complete
+      def wait_by_id(node_id, task_id, timeout = "10s", params = {})
+        raise IllegalArgument, "no node ID provided" if node_id.to_s.empty?
+        raise IllegalArgument, "no task ID provided" unless task_id.to_i
+
+        params_with_wait = params.merge({ :wait_for_completion => true, :timeout => timeout })
+        self.get_by_id(node_id, task_id, params_with_wait)
+      end
+
+      # Cancels a task running on a particular node.
+      # NOTE: the API docs note the behavior wrong for this call; "task_id:<task_id>" is really "<node_id>:<task_id>"
+      # where "node_id" is a value from the "nodes" hash returned from the /_tasks endpoint, and "task_id" is
+      # from the "tasks" child hash of any of the "nodes" entries of the /_tasks endpoint
+      #
+      # node_id         - the ES node hosting the task to be cancelled
+      # task_id         - ID of the task to be cancelled
+      # params          - Hash of request parameters to include
+      #
+      # Examples
+      #
+      #   tasks.cancel_by_id "DmteLdw1QmSgW3GZmjmoKA", 123
+      #   tasks.cancel_by_id "DmteLdw1QmSgW3GZmjmoKA", 456, :pretty => true
+      #
+      # Returns the response body as a Hash
+      def cancel_by_id(node_id, task_id, params = {})
+        raise IllegalArgument, "no node ID provided" if node_id.to_s.empty?
+        raise IllegalArgument, "no task ID provided" unless task_id.to_i
```

Comment:
The `to_i` method in Ruby works for **everything**.

```ruby
nil.to_i  #=> 0
```

If you want to ensure that the user is passing in a Numeric value, then you can check the type of the `task_id`.

```ruby
raise ArgumentError, "invalid task ID provided: #{task_id.inspect}" unless task_id.is_a?(Integer)
```

The `ArgumentError` here is a Ruby defined exception signaling that the wrong arguments were passed to the method. It is more suitable in this case.

Reply:
Thank makes sense, thanks, will switch to `Numeric` type check. Was hoping to do something that allows for `"123"` and `123` but not `"joe"` will think on it.

Reply:
You can use `Integer()` for that:

```
irb(main):001:0> Integer("1")
=> 1
irb(main):002:0> Integer("1.0")
ArgumentError: invalid value for Integer(): "1.0"
	from (irb):2:in `Integer'
	from (irb):2
	from /Users/look/.rbenv/versions/2.4.2/bin/irb:11:in `<main>'
irb(main):003:0> Integer(1)
=> 1
irb(main):004:0> Integer("joe")
ArgumentError: invalid value for Integer(): "joe"
	from (irb):4:in `Integer'
	from (irb):4
	from /Users/look/.rbenv/versions/2.4.2/bin/irb:11:in `<main>'
```